src/api/app/controllers/webui/request_controller.rb

```diff
@@ -519,9 +519,14 @@ def staging_status(request, target_project)
   end
 
   def cache_diff_data
-    return unless @action[:diff_not_cached]
+    bs_request_action = @action
+
+    # [DEPRECATED] TODO: drop the first part of the conditional after the request_workflow_redesign rollout
+    # we need this because before the beta version @action was still an Hash. With the beta it is a pure BsRequestAction instead
+    bs_request_action = BsRequestAction.find(@action[:id]) if @action.instance_of?(Hash)
```

Comment:
```suggestion
    bs_request_action = BsRequestAction.find(@action.id) if @action.instance_of?(Hash)
```

Reply:
Nope, the point is exactly that if the `@action` is an `Hash` we still need to fetch the `id` with `@action[:id]` because this method is used by both beta and non beta version, as the comment above describes it

Reply:
Yeah, I'm fine with that. I mean it's one line marked by the comment, but isolating the code path sounds safer.

Reply:
Done

Reply:
Okay, sorry for the noise, I made a mistake: `cache_diff_data` is used only in the beta version. Refactoring....... :sweat_smile: